internal/keycloakb/eventsdbmodule.go

```diff
@@ -169,6 +191,110 @@ func (cm *eventsDBModule) GetTotalConnectionsCount(_ context.Context, realmName
 	return res, err
 }
 
+// GetTotalConnectionsHoursCount gets the number of connections for the given realm for the last 24 hours, hour by hour
+func (cm *eventsDBModule) GetTotalConnectionsHoursCount(_ context.Context, realmName string) ([][]int64, error) {
+
+	var err error
+	var nbHours = 24
+	var nbConnections = int64(0)
+	var res = make([][]int64, nbHours)
+	for i := 0; i < nbHours; i++ {
+		res[nbHours-i-1] = make([]int64, 2)
+		var row = cm.db.QueryRow(strings.ReplaceAll(selectConnectionsHoursCount, "##OFFSET##", strconv.Itoa(i)), realmName)
+		err = row.Scan(&nbConnections)
+		if time.Now().UTC().Hour()-(i+1) < 0 {
```

Comment:
I guess you can replace this :
```
if time.Now().UTC().Hour()-(i+1) < 0 {
    res[nbHours-i-1][0] = int64(24 + (time.Now().UTC().Hour() - i))
} else {
    res[nbHours-i-1][0] = int64((time.Now().UTC().Hour() - i))
}
```
by
```
    res[nbHours-i-1][0] = int64((nowHour - i + 240)%24)
```
This way, you might be able to factorize execution of selectConnectionsHoursCount, selectConnectionsDaysCount and selectConnectionsMonthsCount

Reply:
Ashamed I did not see this :( 
But I am just wondering...why 240 and not 24 ?

Reply:
240 ? I could have use 504... or any multiple number of 24: it won't change anything with module 24...
In fact, I was thinking to a big multiple of 24 in case you wanted to get statistics on more than 24 hours. (48 or 72 hours)